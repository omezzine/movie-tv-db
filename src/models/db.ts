export type DbType = 'movie' | 'tv';
