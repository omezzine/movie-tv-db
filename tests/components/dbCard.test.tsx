import { describe, expect, test, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import DbCard from '../../src/components/dbCard';
import MockReponse from '../mocks/responseMovieApi.json';
import React from 'react';
import { Movie } from '../../src/models/movie';
import renderer from 'react-test-renderer';

const movieMock: Movie = MockReponse.results[0] as Movie;

describe('DB Card test', () => {
  test('Should match snapshot', () => {
    const dbCard = renderer.create(<DbCard data={movieMock} type="movie" onClick={vi.fn()} />).toJSON();
    expect(dbCard).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    render(<DbCard data={movieMock} type="movie" onClick={vi.fn()} />);
    expect(screen.getByText(/^The Matrix/i)).toBeDefined();
    expect(screen.getByText(/^Set in the 22nd/i)).toBeDefined();
  });

  test('Should render empty image if no image available', () => {
    const movieMockwithoutiMage: Movie = MockReponse.results[1] as Movie;
    const { container } = render(<DbCard data={movieMockwithoutiMage} type="movie" onClick={vi.fn()} />);
    expect(container.querySelector('img')?.getAttribute('src')).toContain('No-Image-Placeholder.svg.png');
  });

  test('Should call callback when clicked', () => {
    const onCardClick = vi.fn();
    const { container } = render(<DbCard data={movieMock} type="movie" onClick={onCardClick} />);
    fireEvent.click(container.querySelector('.card') as HTMLElement);
    expect(onCardClick).toHaveBeenCalledWith({ id: movieMock.id, type: 'movie' });
  });
});
