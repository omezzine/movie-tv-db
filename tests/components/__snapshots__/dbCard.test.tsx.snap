// Vitest Snapshot v1

exports[`DB Card test > Should match snapshot 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <img
    alt="Card image cap"
    className="card-img-top"
    src="https://image.tmdb.org/t/p/w500/waCRuAW5ocONRehP556vPexVXA9.jpg"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      The Matrix
    </h5>
    <p
      className="card-text text-truncate"
    >
      Set in the 22nd century, The Matrix tells the story of a computer hacker who joins a group of underground insurgents fighting the vast and powerful computers who now rule the earth.
    </p>
  </div>
</div>
`;
